packages/next/src/build/webpack/plugins/slow-module-detection-plugin.ts

```diff
@@ -0,0 +1,255 @@
+import type { Compiler, Module, Compilation } from 'webpack'
+import type { CompilerNameValues } from '../../../shared/lib/constants'
+import { getModuleBuildInfo } from '../loaders/get-module-build-info'
+import { yellow, green, blue } from '../../../lib/picocolors'
+
+const PLUGIN_NAME = 'SlowModuleDetectionPlugin'
+
+const TreeSymbols = {
+  VERTICAL_LINE: '│  ',
+  BRANCH: '├─ ',
+} as const
+
+const DEFAULT_SLOW_MODULE_THRESHOLD_MS = 500
+
+const DEFAULT_PATH_TRUNCATION_LENGTH = 100
+
+// Matches node_modules paths, including pnpm-style paths
+const NODE_MODULES_PATH_PATTERN = /node_modules(?:\/\.pnpm)?\/(.*)/
+
+interface ModuleBuildTimeAnalyzerOptions {
+  compilerType: CompilerNameValues
+  slowModuleThresholdMs?: number
+  pathTruncationLength?: number
+}
+
+const getModuleIdentifier = (module: Module): string => {
+  const debugId = module.debugId
+  if (!debugId) {
+    throw new Error(
```

Comment:
we already return when moduleId is not found in modules, maybe just return sth else here instead of throwing an error? I feel it's safe to skip this case as long as the main feature is working